modules/nf-scil/preproc/topup/main.nf

```diff
@@ -0,0 +1,96 @@
+
+process PREPROC_TOPUP {
+    tag "$meta.id"
+    label 'process_single'
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
+        'scilus/scilus:1.6.0' }"
+
+    input:
+        tuple val(meta), path(dwi), path(bval), path(bvec), path(b0), path(rev_dwi), path(rev_bval), path(rev_bvec), path(rev_b0)
+        path(config_topup)
+
+    output:
+        tuple val(meta), path("*__corrected_b0s.nii.gz"), emit: topup_corrected_b0s
+        tuple val(meta), path("*_fieldcoef.nii.gz"), emit: topup_fieldcoef
+        tuple val(meta), path("*_movpar.txt"), emit: topup_movpart
+        tuple val(meta), path("*__rev_b0_warped.nii.gz"), emit: rev_b0_warped
+        tuple val(meta), path("*__rev_b0_mean.nii.gz"), emit: rev_b0_mean
+        tuple val(meta), path("*__b0_mean.nii.gz"), emit: b0_mean
+        path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def prefix_topup = task.ext.prefix_topup ? task.ext.prefix_topup : ""
+    def config_topup = config_topup ?: task.ext.default_config_topup
+    def encoding = task.ext.encoding ? task.ext.encoding : ""
+    def readout = task.ext.readout ? task.ext.readout : ""
+    def b0_thr_extract_b0 = task.ext.b0_thr_extract_b0 ? task.ext.b0_thr_extract_b0 : ""
```

Comment:
For your knowledge, and if you want to do it here, there is a shorter way to write the ternary operator when the quantities around the `?` are the same, e.g. :

```
    (task.ext.prefix_topup ? task.ext.prefix_topup : "") === (task.ext.prefix_topup ?: "")
```

Just a bit of nice groovy flavour :stuck_out_tongue: